python/cuml/utils/input_utils.py

```diff
@@ -223,22 +188,28 @@ def input_to_dev_array(X, order='F', deepcopy=False,
                           "but got the opposite. Converting data, this will "
                           "result in additional memory utilization.")
             X_m = rmm_cupy_ary(cp.array, X_m, copy=False, order=order)
-            X_m = cuda.as_cuda_array(X_m)
+            X_m = Array(data=X_m)
 
-    X_ptr = get_dev_array_ptr(X_m)
+    ptr = X_m.ptr
+    if legacy:
+        X_m = cuda.as_cuda_array(X_m)
 
-    return inp_array(array=X_m, pointer=X_ptr, n_rows=n_rows, n_cols=n_cols,
-                     dtype=dtype)
+    return inp_array(array=X_m, pointer=ptr, n_rows=n_rows, n_cols=n_cols,
```

Comment:
Is it even necessary to return the pointer in the named tuple in the future? With the new Array, it seems easy enough for callers to access all of this info so we could simplify the API a bit.

Reply:
I didn't change it to keep this as the same function as it was above, but with the suggestion of it being a new function I think its better to not return it